api/pkg/user/service/user.go

```diff
@@ -0,0 +1,245 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/app"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"github.com/tektoncd/hub/api/pkg/token"
+	userApp "github.com/tektoncd/hub/api/pkg/user/app"
+	"gorm.io/gorm"
+)
+
+type request struct {
+	db            *gorm.DB
+	log           *log.Logger
+	user          *model.User
+	defaultScopes []string
+	jwtConfig     *app.JWTConfig
+}
+
+type UserService struct {
+	app.Service
+	api       app.Config
+	JwtConfig *app.JWTConfig
+}
+
+type Service interface {
+	Info(res http.ResponseWriter, req *http.Request)
+	RefreshAccessToken(res http.ResponseWriter, req *http.Request)
+	NewRefreshToken(res http.ResponseWriter, req *http.Request)
+}
+
+var (
+	invalidRefreshToken = fmt.Errorf("invalid refresh token")
+	refreshError        = fmt.Errorf("failed to refresh token")
+)
+
+// New returns the auth service implementation.
+func New(api app.Config) Service {
+	return &UserService{
+		Service:   api.Service("user"),
+		api:       api,
+		JwtConfig: api.JWTConfig(),
+	}
+}
+
+// Get the user Info
+func (s *UserService) Info(res http.ResponseWriter, req *http.Request) {
```

Comment:
should we validate the access token here ?

Reply:
Since we are using a middleware it  verifies and validates the access token whenever a request is made to the endpoint